src/main/java/com/cloudbees/jenkins/support/BundleNameInstanceTypeProvider.java

```diff
@@ -0,0 +1,78 @@
+package com.cloudbees.jenkins.support;
+
+import com.cloudbees.jenkins.support.api.SupportProvider;
+import com.google.common.annotations.VisibleForTesting;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+
+import javax.annotation.Nonnull;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * <p>Extension point allowing to customize the support bundle naming strategy.</p>
+ * <p>
+ * It will work the following way:
+ * </p>
+ * <ol>
+ * <li>If an implementation of {@link BundleNameInstanceTypeProvider} is found, it will be used.<br>
+ * <strong>WARNING: </strong>if many are found, then a warning will be issued, and the first extension found will
+ * be used.</li>
+ * <li>If not, then it will check for the presence of the {@link #SUPPORT_BUNDLE_NAMING_INSTANCE_SPEC_PROPERTY}
+ * system property, and will use its value if provided.</li>
+ * <li>If not, then will fallback to the original behaviour, which is simply an empty String</li>
+ * </ol>
+ *
+ * @see SupportProvider#getName() for prefixing.
+ */
+public abstract class BundleNameInstanceTypeProvider implements ExtensionPoint {
+
+    @VisibleForTesting
+    static final String SUPPORT_BUNDLE_NAMING_INSTANCE_SPEC_PROPERTY = SupportPlugin.class.getName() + ".instanceType";
+
+    private static final Logger LOGGER = Logger.getLogger(BundleNameInstanceTypeProvider.class.getName());
+
+    private static final BundleNameInstanceTypeProvider DEFAULT_STRATEGY = new BundleNameInstanceTypeProvider() {
+
+        @Override
+        public String getInstanceType() {
+            return System.getProperty(SUPPORT_BUNDLE_NAMING_INSTANCE_SPEC_PROPERTY, "");
+        }
+    };
+
+    /* package */
+    @Nonnull
+    static BundleNameInstanceTypeProvider getInstance() {
+        final ExtensionList<BundleNameInstanceTypeProvider> all = ExtensionList.lookup(BundleNameInstanceTypeProvider.class);
+        final int extensionCount = all.size();
+        if (extensionCount < 1) {
+            LOGGER.fine("No alternative strategy provided for support bundle prefixing.");
+            return DEFAULT_STRATEGY;
+        }
+
+        if (all.size() > 1) {
+            if (LOGGER.isLoggable(Level.WARNING)) {
+                LOGGER.warning("{0} implementations found for support bundle prefix naming strategy. " +
```

Comment:
:bug: you are missing a parameter here.
NIT: from a stylistic point I think its better to always use Logger.log(Level.XXX ...) rather than the Logger.xxx(...)